js/assessment.js

```diff
@@ -291,12 +293,19 @@ define([
             this._assessments.push(assessmentModel);
 
             this._restoreModelState(assessmentModel);
+            this._restoredCount++;
 
             Adapt.trigger('assessments:register', state, assessmentModel);
 
             this._setPageProgress();
 
             this._setupQuestionNumbering();
+
+            if (this._restoredCount === this._assessments.length) {
```

Comment:
just a suggestion but this could be changed to `if (++this._restoredCount === this._assessments.length)` and then you could do away with L296